test/cypress/integration/agreements/agreement.cy.js

```diff
@@ -0,0 +1,78 @@
+const userStubs = require('../../stubs/user-stubs')
+const gatewayAccountStubs = require('../../stubs/gateway-account-stubs')
+const agreementStubs = require('../../stubs/agreement-stubs')
+const transactionStubs = require('../../stubs/transaction-stubs')
+
+const userExternalId = 'some-user-id'
+const gatewayAccountId = 10
+const gatewayAccountExternalId = 'gateway-account-id'
+const serviceExternalId = 'service-id'
+
+const userAndGatewayAccountStubs = [
+  userStubs.getUserSuccess({ userExternalId, serviceExternalId, gatewayAccountId }),
+  gatewayAccountStubs.getGatewayAccountByExternalIdSuccess({
+    gatewayAccountId,
+    gatewayAccountExternalId,
+    serviceExternalId,
+    recurringEnabled: true
+  })
+]
+
+const transactionsStub = transactionStubs.getLedgerTransactionsSuccess({
+  gatewayAccountId,
+  transactions: [
+    { reference: 'payment-reference', amount: 1000, type: 'payment' },
+    { reference: 'second-reference', amount: 20000, type: 'payment' }
+  ],
+  filters: {
+    agreement_id: 'a-valid-agreement-id',
+    display_size: 5
+  }
+})
+
+describe('Agreements', () => {
```

Comment:
Could increase test coverage here. ex:
- the user does not have permission to cancel the agreement
- when cancel fails ( connector returns error) 

Can be in a separate PR